install/playbooks/roles/borg-backup/files/backup.py

```diff
@@ -151,55 +151,40 @@ def loadKey(self, path):
     def mountRepository(self):
         """Mount the remote location if necessary"""
 
-        ## When using SSH, we do not need to mount remote SSH location.
-        # for this scheme, it is assumed that the remote server has borg installed
-        # otherwise, use sshfs://
+        # When using SSH, we do not need to mount remote SSH location.
+        # for this scheme, it is assumed that the remote server has borg
+        # installed otherwise, use sshfs://
         if self.location.scheme == 'ssh':
-            self.repositoryPath = self.url[6:]
-            self.repositoryMounted = True
+            self.repositoryPath = self.url + '/@server'
+            self.repositoryMounted = False
+            return True
+
+        # The dir:// backup location can be a remote directory mounted locally
+        # or even a local partition.
+        if self.location.scheme == 'dir':
+            self.repositoryPath = self.location.path + '/@server'
+            self.repositoryMounted = False
             return True
 
         # Make sure the directory to mount the backup exists
         self.mountPath = '/mnt/backup/' + self.configName
         os.makedirs(self.mountPath, exist_ok=True)
 
-        # The backup location can be a remote directory mounted locally
-        # or even a local partition. Removing dir://
-        if self.location.scheme == 'dir':
-            self.repositoryPath = self.location.path
-            self.repositoryMounted = True
-            return True
-
-        # The location is a USB device mounted automatically using systemd
-        if self.location.scheme == 'usb':
-            self.repositoryPath = self.location.path
-            self.mountPath = '/mnt/backup/' + self.configName
+        # These locations are mounted automatically using systemd
+        if self.location.scheme in {'cifs', 'sshfs'}:
+            # The target directories are directly mounted at the mountPath.
+            self.repositoryPath = self.mountPath + '/@server'
             self.repositoryMounted = os.path.ismount(self.mountPath)
             return self.repositoryMounted
 
-        # The location is an S3 bucket mounted automatically using systemd
-        if self.location.scheme == 's3fs':
-            self.repositoryPath = self.location.path
-            self.mountPath = '/mnt/backup/' + self.configName
+        # These locations are mounted automatically using systemd
+        if self.location.scheme in {'s3fs', 'usb'}:
+            # The root of the remote filesystem is mounted at the mountPath,
+            # add the location path part to reflect the given URL.
+            self.repositoryPath = self.mountPath + self.location.path + '/@server'
```

Comment:
This changes the way URLs are configured for `usb` and `s3fs` (cf. doc changes). The `/mnt/backup/<location.name>` could be stripped to support old confs, but maybe it would be better to advertise for it and let it break if unchanged (nonexistent directory), so that people upgrade the URLs.